src/v/cluster/scheduling/leader_balancer_greedy.h

```diff
@@ -53,7 +56,7 @@ class greedy_balanced_shards final : private leader_balancer_strategy {
     double adjusted_error(
       const model::broker_shard& from, const model::broker_shard& to) const {
         auto target_load = static_cast<double>(_num_groups)
-                           / static_cast<double>(_cores.size());
+                           / static_cast<double>(_num_cores);
```

Comment:
So this seems a bit suspicious to me because the target load is unattainable if some nodes are muted, but retain leadership. In this case we'll try to move that leaders from muted nodes to non-muted ones and fail. E.g. 5 nodes (n1, n2, n3, n4, n5), topic replication factor 3, n1 is the controller leader, n4 and n5 are down or partitioned. In this case n4 and n5 will still retain some leadership which we'll try to move from them (and fail and mute the corresponding groups).

Maybe the simpler and easier to reason about heuristic is to simply exclude muted nodes from consideration and try to move leaders between the non-muted nodes. In this case `_num_groups` becomes "number of leaders on non-muted nodes", muted nodes are completely excluded from index, and we simply try to rebalance leaders on the reachable nodes (which is going to have much higher chance to succeed).

@dotnwat what do you think?

Reply:
Yes, this is a mistake. I had always intended to not move _from_ muted nodes as well as not move _to_ muted nodes. Good catch!